### PR TITLE
Add type-less overload for Parse method #726

### PR DESCRIPTION
**Pull request by gstoian**
Made dataType parameter optional in FhirXmlParser. Added Unit test for this change.

### CODE REVIEW

File: src/Hl7.Fhir.Core.Tests/Serialization/SerializationTests.cs [hunk 1]
```diff
@@ -53,6 +53,20 @@ public void ParseMetaXml()
             Assert.AreEqual(metaXml, xml);
         }
 
+        [TestMethod]
+        public void ParsePatientXmlNullType()
+        {
+            string xmlPacientTest = TestDataHelper.ReadTestData("TestPatient.xml");
+            
+            var poco = new FhirXmlParser().Parse(xmlPacientTest);
+            var xml = new FhirXmlSerializer().SerializeToString(poco);
```
Comment (ewoutkramer) [inline src/Hl7.Fhir.Core.Tests/Serialization/SerializationTests.cs]: I think we can remove the `var xml....` line, it does not do anything anymore?

File: src/Hl7.Fhir.Core.Tests/Serialization/SerializationTests.cs [hunk 2]
```diff
@@ -631,18 +645,18 @@ public void SummarizeSerializingTest()
         public void DateTimeOffsetAccuracyTest()
```
Comment (ewoutkramer) [inline src/Hl7.Fhir.Core.Tests/Serialization/SerializationTests.cs]: I think these changes were not supposed to be part of this PR?

File: src/Hl7.Fhir.Core/Serialization/FhirXmlParser.cs [hunk 3]
```diff
@@ -30,7 +30,7 @@ public FhirXmlParser(ParserSettings settings=null) : base(settings)
                     DisallowSchemaLocation = Settings.DisallowXsiAttributesOnRoot,
                 };
 
-        public Base Parse(string xml, Type dataType)
+        public Base Parse(string xml, Type dataType = null)
```
Comment (ewoutkramer) [inline src/Hl7.Fhir.Core/Serialization/FhirXmlParser.cs]: There is another overload on Parse() which takes an XmlReader, and is also missing the `= null`.  The same is true for these functions on the FhirJsonParser.cs

File: src/Hl7.Fhir.Core.Tests/Serialization/SerializationTests.cs (hunk 2 above)
Comment (gstoian) [reply src/Hl7.Fhir.Core.Tests/Serialization/SerializationTests.cs]: Yes, they were not. My visual studio doesn't see them at all.

**Pull request merged**